### PR TITLE
configure sanitized build

### PR DESCRIPTION
**Pull request by kyessenov**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kyessenov** [comment]
Missing ncurses https://github.com/istio/tools/pull/2644

**Review by lei-tang — APPROVED** [review]

**Comment by kyessenov** [comment]
ARM should be handled by https://github.com/istio/tools/pull/2646 - @ericvn I'm not seeing the CI image updated yet. How do I kick the CI to pick up the latest image?

File: Makefile.core.mk [hunk 1]
```diff
@@ -100,10 +84,9 @@ gen-check:
 	@scripts/gen-testdata.sh -c
 
 test:
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && \
-	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT) $(TEST_ENVOY_TARGET)
+	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT) -- $(TEST_ENVOY_TARGET) $(BAZEL_TEST_TARGETS)
 	if [ -n "$(BAZEL_TEST_TARGETS)" ]; then \
-	  export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT) $(BAZEL_TEST_ARGS) -- $(BAZEL_TEST_TARGETS); \
```
Comment (kyessenov) [inline Makefile.core.mk:106]: FYI Path is being removed from bazel sandbox gradually to improve hermeticity.

**Comment by ericvn** [comment]
> ARM should be handled by [istio/tools#2646](https://github.com/istio/tools/pull/2646) - @ericvn I'm not seeing the CI image updated yet. How do I kick the CI to pick up the latest image?

@kyessenov Because the latest image doesn't work (nor the one before it). I created https://github.com/istio/proxy/pull/4926 to test the new image and it's still failing on the earlier change.

**Comment by kyessenov** [comment]
@ericvn Yes, we have to bump them together. How do I bump tools image?

**Comment by ericvn** [comment]
> @ericvn Yes, we have to bump them together. How do I bump tools image?

We can manually do the proxy pipeline, or just bump all repos. Since it looks like we are farther, I'll kick off the update to the latest tools image.

**Pull request merged**